Dockerfile

```diff
@@ -1,49 +1,37 @@
-FROM node:16-slim
+#
+# Build
+#
+FROM node:16-slim AS builder
 
-# Install node_modules into a different directory to avoid npm/npm#9863.
-RUN mkdir -p /srv/node
-COPY package.json /srv/node/
-COPY yarn.lock /srv/node/
 WORKDIR /srv/node
+COPY package.json yarn.lock /srv/node/
+
+RUN yarn install --pure-lockfile
+
+#
+# Install
+#
+FROM node:16-slim
+
+ARG app_uid=9500
+ARG app_dir=/app
+
+RUN useradd -u ${app_uid} -d /home/app -m -s /sbin/nologin app
+# The WORKDIR directive set the ownership of the work directory to root instead
+# of USER unless the "buildkit" feature is enabled. To make sure the work
+# directory is owned by the proper user for everybody, we manually set the
+# ownership.
+RUN mkdir -p ${app_dir} && chown ${app_uid}:${app_uid} ${app_dir}
+
+USER ${app_uid}:${app_uid}
+
+WORKDIR ${app_dir}
 
-# This file has been downloaded from: https://dl.yarnpkg.com/debian/pubkey.gpg
-COPY docker/etc/pki/yarnpkg.gpg.key /etc/pki/yarnpkg.gpg.key
```

Comment:
If this isn't useful anymore, this should be removed too.

Reply:
Removed. Thanks!